src/contentScripts/other/selma/layout.ts

```diff
@@ -0,0 +1,500 @@
+      if (scriptElm === null) continue
+
+      const scriptContent = scriptElm!.innerHTML
+
+      const url = scriptToURL(scriptContent)
+
+      promises.push(
+        fetch(url).then(async (s) => {
+          const parser = new DOMParser()
+          const doc = parser.parseFromString(await s.text(), 'text/html')
+
+          return { doc, elm: lastCol, url }
+        })
+      )
+    }
+
+    promises.forEach((p) =>
+      p.then(({ doc, elm, url }) => {
+        const tableBody = doc.querySelector('tbody')!
+        const values = [...tableBody.children].map((tr) => {
+          const gradeText = tr.children.item(0)!.textContent!.replace(',', '.')
+          const grade = parseFloat(gradeText)
+
+          const countText = tr.children.item(1)!.textContent!
+          let count: number
+          if (countText === '---') count = 0
+          else count = parseInt(countText)
+
+          return {
+            grade,
+            count
+          }
+        })
+        // .slice(0, -2); // Remove the 5.0 from all lists
+
+        // Present the bar chart
+        const graphSVG = Graphing.createSVGGradeDistributionGraph(values, url)
+        elm.innerHTML = graphSVG
+      })
+    )
+
+    // Remove the inline style that sets a width on the top right table cell
+    const tableHeadRow = document.querySelector('thead>tr')!
+    tableHeadRow.children.item(3)!.removeAttribute('style')
+    /*
+
+*/
+  } else if (currentView.startsWith('/APP/COURSERESULTS/')) {
+    // Prüfungen > Ergebnisse
+
+    // Remove the "bestanden" section
+    const headRow = document.querySelector('thead>tr')!
+    headRow.removeChild(headRow.children.item(3)!)
+
+    // Add "Notenverteilung" header
+    {
+      headRow.children.item(3)!.removeAttribute('colspan')
+      const newHeader = document.createElement('th')
+      newHeader.textContent = 'Notenverteilung'
+      headRow.appendChild(newHeader)
+    }
+
+    // Create the grade distribution graph
+    const body = document.querySelector('tbody')!
+    const promises: Promise<{ doc: Document; elm: Element; url: string }>[] =
+      []
+    for (const row of body.children) {
+      // Remove useless inline styles which set the vertical alignment
+      for (const col of row.children) col.removeAttribute('style')
+
+      // Remove "Status" column
+      row.removeChild(row.children.item(3)!)
+
+      {
+        // Map grade descriptions to emojis
+        const gradeElm = row.children.item(2)!
+        mapGrade(gradeElm)
+      }
+
+      // Extract script content
+      const lastCol = row.children.item(4)!
+      const scriptElm = lastCol.children.item(1)
+      // Skip courses wihtout grades
+      if (scriptElm === null) continue
+
+      const scriptContent = scriptElm!.innerHTML
+
+      const url = scriptToURL(scriptContent)
+
+      promises.push(
+        fetch(url).then(async (s) => {
+          const parser = new DOMParser()
+          const doc = parser.parseFromString(await s.text(), 'text/html')
```

Comment:
The page contains script elements diretly after the <a> link elements that contain the code to open the Popup, using a RegExp the arguments for this URL get extracted from the script ot then construct the URL that would get opened in the PopUp
![image](https://github.com/user-attachments/assets/9e789179-d2db-4172-8907-1027adbc3160)

Reply:
Extracting the individual tries works pretty much the same, extracting the information is just a bit more complicated and unreliable. I am not 100% happy with how it is working right now as it still has trouble with some modules that format their information tables differently. For example language courses don't work as of now as shown in the picture below. Solving this issue definitely is not hard I just do not have the time right now :)

![image](https://github.com/user-attachments/assets/5bb94478-c6ba-4170-89ab-db20f3105f11)

The individual tries also show the grade and date when hovering over them, the grade distribution charts also have tooltips showing what grade the bar represents. All graphics still act as clickable button that opens the original link just like before, though I am not sure if this immediately obvious and there might be a better solution to indicate to users that the charts are clickable